### PR TITLE
InfluxDB: Escape and interpolate tag values

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

When a template variable is used for tag value we interpolate it. But if the variable has special characters then the query is broken. This PR is addressing that issue.

Backend migrated version of this https://github.com/grafana/grafana/pull/39170

Part of Epic: https://github.com/grafana/grafana/issues/65045

**Why do we need this feature?**

To have better interpolation on tag values

**Who is this feature for?**

InfluxDB users with the feature flag influxdbBackendMigration enabled.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

### How to test
- Have a variable that holds a value with a special character like "." i.e  ($variable = test.test)
- Use that variable in a query.
- Without the feature toggle it works but with the feature toggle, we were escaping the dot in value. 
- This  PR is fixing that issue.
- Use case https://github.com/grafana/support-escalations/issues/5317

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -311,9 +311,15 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
 
     if (query.tags) {
       expandedQuery.tags = query.tags.map((tag) => {
+        let val = tag.value;
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/influxdb/datasource.ts:314]: Here's a unit test for this: 

https://github.com/grafana/grafana/pull/70101/files#diff-62cbde0cbe63fcdf60ae080a5f796eda77d9529ba6f9bfc1da9a618e735b047cR161-R179

**Comment by gabor** [comment]
hi @itsmylife , the diff looks reasonable, but i cannot seem to reproduce the problem.
could you please describe a test-scenario (step by step), that fails on main-branch grafana, and works in this PR? thanks!

**Review by LudoVio — APPROVED** [review]

**Comment by itsmylife** [comment]
> hi @itsmylife , the diff looks reasonable, but i cannot seem to reproduce the problem. could you please describe a test-scenario (step by step), that fails on main-branch grafana, and works in this PR? thanks!

Hi Gabor, Here is the case (I will add this to description too.)
- Have a variable which holds a value with a special character like "." i.e  ($variable = test.test)
- Use that variable in a query.
- Without feature toggle it works but with feature toggle we were escaping the dot in value. 
- This  PR is fixing that issue.
- Use case https://github.com/grafana/support-escalations/issues/5317

**Comment by gabor** [comment]
@itsmylife thanks for the how-to-test info...

i did some testing, and...

- i inserted the following measurement into influxdb: `"m5,tag1=va.l5 val=45"`
   - so the tag-name is `tag1` and tag-value is `va.l5`
- i created a dashboard, i created a custom variable where one of it's valus was `va.l5`, and created an influxdb-influxql-query that uses that variable in the `WHERE` section. then i ran the query, and checked what is sent to the infludb-database both on main-branch grafana, and on this PR:
    - main-branch:
         - `WHERE "tag1"::tag =~ /^va\.l5$/`
    - this PR:
         - `WHERE "tag1"::tag =~ /^va.l5$/`

i think the main-branch's escaping is more correct. the regular expression of `/va.l5/` matches any character in place of the `.`, so it will match things like `vaXl5` or `va_l5` and also `va.l5`.

**Comment by itsmylife** [comment]
@gabor did you use the feature toggle `influxdbBackendMigration`?

**Pull request closed without merging**